roofit/roofit/test/testRooPolyFunc.cxx

```diff
@@ -0,0 +1,77 @@
+// Author: Rahul Balasubramanian, CERN  12/2021
+
+#include "RooPolyFunc.h"
+
+#include "RooPolynomial.h"
+#include "RooRealVar.h"
+#include "RooFormulaVar.h"
+#include "RooWrapperPdf.h"
+
+#include <numeric>
+#include <string>
+#include <iostream>
+
+#include "gtest/gtest.h"
+
+#define MAKE_POLYFUNC_AND_POLYPDF                            \
+   RooRealVar x("x", "x", 0., -200., 200.);                  \
+   RooRealVar c0("c0", "c0", 0., 1.E-6, 100.);               \
+   RooRealVar c1("c1", "c1", 0., 1.E-6, 100.);               \
+   RooRealVar c2("c2", "c2", 0., 1.E-6, 100.);               \
+   RooPolyFunc polyfunc("f", "f", RooArgSet(x));             \
+   polyfunc.addTerm(c0.getVal(), x, 0);                      \
+   polyfunc.addTerm(c1.getVal(), x, 1);                      \
+   polyfunc.addTerm(c2.getVal(), x, 2);                      \
+   RooWrapperPdf wrapperpdf("wrappdf", "wrappdf", polyfunc); \
+   RooPolynomial pdf("pdf", "pdf", x, RooArgList(c0, c1, c2));
+
+#define MAKE_POLYFUNC_AND_POLYFORMULA            \
+   RooRealVar x("x", "x", 0., -200., 200.);      \
+   RooRealVar c0("c0", "c0", 0., 1.E-6, 100.);   \
+   RooRealVar c1("c1", "c1", 0., 1.E-6, 100.);   \
+   RooRealVar c2("c2", "c2", 0., 1.E-6, 100.);   \
+   RooPolyFunc polyfunc("f", "f", RooArgSet(x)); \
+   polyfunc.addTerm(c0.getVal(), x, 0);          \
+   polyfunc.addTerm(c1.getVal(), x, 1);          \
+   polyfunc.addTerm(c2.getVal(), x, 2);          \
+   RooFormulaVar formula("formula", "formula", "@1 + @2*pow(@0,1) + @3*pow(@0,2)", RooArgList(x, c0, c1, c2));
+
+TEST(RooPolyFunc, WrappedPdfClosure)
+{
+   MAKE_POLYFUNC_AND_POLYPDF
+   for (double theX : {-100., -50., -10., -1., 0., 1., 10., 50., 100.}) {
+      for (double theC0 : {0., 2.5, 5, 7.5, 10.}) {
+         for (double theC1 : {0., 2.5, 5, 7.5, 10.}) {
+            for (double theC2 : {0., 2.5, 5, 7.5, 10.}) {
+               x = theX;
+               c0 = theC0;
+               c1 = theC1;
+               c2 = theC2;
+               // wrapped pdf of RooPolyFunc should match RooPolynomial
+               EXPECT_FLOAT_EQ(wrapperpdf.getVal(), pdf.getVal())
+                  << theX << " " << theC0 << " " << theC1 << " " << theC2;
+            }
+         }
+      }
+   }
+}
+
+TEST(RooPolyFunc, FormulaVarClosure)
+{
+   MAKE_POLYFUNC_AND_POLFORMULA
+   for (double theX : {-100., -50., -10., -1., 0., 1., 10., 50., 100.}) {
+      for (double theC0 : {0., 2.5, 5, 7.5, 10.}) {
+         for (double theC1 : {0., 2.5, 5, 7.5, 10.}) {
+            for (double theC2 : {0., 2.5, 5, 7.5, 10.}) {
```

Comment:
```suggestion
   RooFormulaVar formula("formula", "formula", "@1 + @2*pow(@0,1) + @3*pow(@0,2)", RooArgList(x, c0, c1, c2));

using Doubles = std::initializer_list<double>;

TEST(RooPolyFunc, WrappedPdfClosure)
{
   MAKE_POLYFUNC_AND_POLYPDF
   for (double theX : Doubles{-100., -50., -10., -1., 0., 1., 10., 50., 100.}) {
      for (double theC0 : Doubles{0., 2.5, 5, 7.5, 10.}) {
         for (double theC1 : Doubles{0., 2.5, 5, 7.5, 10.}) {
            for (double theC2 : Doubles{0., 2.5, 5, 7.5, 10.}) {
               x = theX;
               c0 = theC0;
               c1 = theC1;
               c2 = theC2;
               // wrapped pdf of RooPolyFunc should match RooPolynomial
               EXPECT_FLOAT_EQ(wrapperpdf.getVal(), pdf.getVal())
                  << theX << " " << theC0 << " " << theC1 << " " << theC2;
            }
         }
      }
   }
}

TEST(RooPolyFunc, FormulaVarClosure)
{
   MAKE_POLYFUNC_AND_POLFORMULA
   for (double theX : Doubles{-100., -50., -10., -1., 0., 1., 10., 50., 100.}) {
      for (double theC0 : Doubles{0., 2.5, 5, 7.5, 10.}) {
         for (double theC1 : Doubles{0., 2.5, 5, 7.5, 10.}) {
            for (double theC2 : Doubles{0., 2.5, 5, 7.5, 10.}) {
```

Unfortunately, this doesn't compile on my system, because the type can't be figured out automatically. Can you please use the `initializer_list` explicitly (same for the other test below)? Also `#include <initializer_list>` for this to work.